### DISCUSSION TITLE
Custom actions are used on blades in custom positions

### DISCUSSION THREAD

**Discussion by pslxx** [filamentphp/filament#2829] [Ideas]
I need some custom actions  but dont show at preset poisition

like this ,i want to click  icon , handle action with form to edit record

![image](https://user-images.githubusercontent.com/8417207/174449175-41d2bb10-0ba6-41c3-9dbe-4fd9b57e1db7.png)

sry my english is shit

**Comment by xiaomlove** [comment]
I found that this does not pop up the modal box, did you solve it please? Only those returned in getActions() that are rendered in the top right corner will pop up.

我发现这样不能弹出模态框，请问有解决方案了吗？只有在 getActions()中返回渲染在右上角的才能弹出。

**Reply by danharrin** [reply]
The actions need registering before they can show modals or anything, thats why we need `getActions()`.

**Reply by pslxx** [reply]
我把action单独复制了一份出来修改了下相关代码,这样可以在任意页面定义action,在需要使用的地方$this->getCachedAction即可

**Comment by xiaomlove** [comment]
@danharrin I found that setting hidden(true) or visible(false) when defining an Action will not show it at the top, but rendering it elsewhere on the page with getCachedAction() is not clickable. I need the Action not to be displayed at the top but rendered normally where I need it.
